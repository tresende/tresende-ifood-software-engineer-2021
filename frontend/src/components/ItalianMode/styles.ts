import { makeStyles } from '@material-ui/core'

const useStyles = makeStyles(() => ({
  title: {
    fontFamily: "'Oleo Script', cursive",
    color: 'white',
    cursor: 'pointer',
    fontSize: '2em'
  }
}))
export default useStyles
