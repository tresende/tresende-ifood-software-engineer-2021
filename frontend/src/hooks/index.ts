export * from './use-ifood-theme'
export * from './use-orders'
