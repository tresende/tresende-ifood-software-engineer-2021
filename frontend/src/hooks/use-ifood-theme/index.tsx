/* eslint-disable @typescript-eslint/no-empty-function */
import { useContext, createContext, useState } from 'react'
import { IFoodTheme, IFoodThemeContextData, IFoodThemeProviderProps } from './types'

const defaultValues: IFoodThemeContextData = {
  changeTheme: () => {},
  italianMode: false
}

export const IFoodThemeContext = createContext<IFoodThemeContextData>(defaultValues)

const IFoodThemeProvider = ({ children }: IFoodThemeProviderProps) => {
  const [theme, setTheme] = useState<IFoodTheme>('default')
  const changeTheme = () => setTheme(theme === 'default' ? 'italian' : 'default')

  const italianMode = theme === 'italian'
  return (
    <IFoodThemeContext.Provider
      value={{
        changeTheme,
        italianMode
      }}
    >
      {children}
    </IFoodThemeContext.Provider>
  )
}

const useIFoodTheme = () => useContext(IFoodThemeContext)

export { IFoodThemeProvider, useIFoodTheme }
