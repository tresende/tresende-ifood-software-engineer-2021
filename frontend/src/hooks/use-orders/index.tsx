/* eslint-disable @typescript-eslint/no-empty-function */
import { useContext, createContext, useState } from 'react'
import { OrdersContextData, OrdersProviderProps } from './types'
import { Filter, Order } from '../../types'
import { SearchOrders } from '../../services/OrderService'

const OrdersContextDefaultValues: OrdersContextData = {
  setFilter: () => {},
  search: () => {},
  orders: []
}

export const OrdersContext = createContext<OrdersContextData>(OrdersContextDefaultValues)

const OrdersProvider = ({ children }: OrdersProviderProps) => {
  const [orders, setOrders] = useState<Order[]>([])
  const [filter, setFilter] = useState<Filter>({})

  const search = async () => {
    try {
      const data = await SearchOrders(filter)
      setOrders(data)
    } catch {
      setOrders([])
    }
  }

  return (
    <OrdersContext.Provider
      value={{
        orders,
        setFilter,
        search
      }}
    >
      {children}
    </OrdersContext.Provider>
  )
}

const useOrders = () => useContext(OrdersContext)

export { OrdersProvider, useOrders }
