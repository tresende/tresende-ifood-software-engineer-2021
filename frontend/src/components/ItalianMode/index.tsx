import { useIFoodTheme } from '../../hooks'
import useStyles from './styles'

const ItalianMode = () => {
  const { changeTheme } = useIFoodTheme()
  const classes = useStyles()
  return (
    <span className={classes.title} onClick={changeTheme}>
      Italian Mode
    </span>
  )
}

export default ItalianMode
