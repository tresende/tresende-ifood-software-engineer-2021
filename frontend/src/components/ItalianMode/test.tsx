import { render } from '@testing-library/react'
import { useIFoodTheme } from 'hooks'

import ItalianMode from '.'

jest.mock('../../hooks')

describe('<ItalianMode />', () => {
  it('should take snapshot with italian mode', () => {
    ;(useIFoodTheme as jest.Mock).mockReturnValue({ italianMode: true })

    const { container } = render(<ItalianMode />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should take snapshot with default mode', () => {
    ;(useIFoodTheme as jest.Mock).mockReturnValue({ italianMode: false })

    const { container } = render(<ItalianMode />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
