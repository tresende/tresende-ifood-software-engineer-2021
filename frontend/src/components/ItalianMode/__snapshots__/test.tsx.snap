// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItalianMode /> should take snapshot with default mode 1`] = `
<span
  class="makeStyles-title-2"
>
  Italian Mode
</span>
`;

exports[`<ItalianMode /> should take snapshot with italian mode 1`] = `
<span
  class="makeStyles-title-1"
>
  Italian Mode
</span>
`;
